lib/Mojolicious/Command/get.pm

```diff
@@ -0,0 +1,231 @@
+    'u|user=s'               => \my $user,
+    'v|verbose'              => \my $verbose;
+
+  @args = map { decode 'UTF-8', $_ } @args;
+  die $self->usage unless my $url = shift @args;
+  my $selector = shift @args;
+
+  # Parse header pairs
+  my %headers = map { /^\s*([^:]+)\s*:\s*(.*+)$/ ? ($1, $2) : () } @headers;
+
+  # Detect proxy for absolute URLs
+  $url !~ m!^/! ? $ua->proxy->detect : $ua->server->app($self->app);
+  $url = Mojo::URL->new($url)->userinfo($user) if $user;
+  $ua->max_redirects(10)                       if $redirect;
+
+  my $buffer = '';
+  $ua->on(
+    start => sub {
+      my ($ua, $tx) = @_;
+
+      # Verbose
+      weaken $tx;
+      $tx->res->content->on(body => sub { warn _header($tx->req), _header($tx->res) }) if $verbose;
+
+      # Stream content (ignore redirects)
+      $tx->res->content->unsubscribe('read')->on(
+        read => sub {
+          return if $redirect && $tx->res->is_redirect;
+          defined $selector ? ($buffer .= pop) : print pop;
+        }
+      );
+    }
+  );
+
+  # Switch to verbose for HEAD requests
+  $verbose = 1 if $method eq 'HEAD';
+  STDOUT->autoflush(1);
+  my @content = %form ? (form => \%form) : defined $in ? ($in) : ();
+  my $tx      = $ua->start($ua->build_tx($method, $url, \%headers, @content));
+  my $res     = $tx->result;
+
+  # JSON Pointer
+  return undef unless defined $selector;
+  return _json($buffer, $selector) if !length $selector || $selector =~ m!^/!;
+
+  # Selector
+  $charset //= $res->content->charset || $res->default_charset;
+  _select($buffer, $selector, $charset, @args);
+}
+
+sub _form { push @{$_[0]{$1}}, $2 ? {file => $3} : $3 }
+
+sub _header { $_[0]->build_start_line, $_[0]->headers->to_string, "\n\n" }
+
+sub _json {
+  return unless my $data = j(shift);
+  return unless defined($data = Mojo::JSON::Pointer->new($data)->get(shift));
+  _say(ref $data eq 'HASH' || ref $data eq 'ARRAY' ? to_json($data) : $data);
+}
+
+sub _say { length && say encode('UTF-8', $_) for @_ }
+
+sub _select {
+  my ($buffer, $selector, $charset, @args) = @_;
+
+  # Keep a strong reference to the root
+  $buffer = decode($charset, $buffer) // $buffer if $charset;
+  my $dom     = Mojo::DOM->new($buffer);
+  my $results = $dom->find($selector);
+
+  while (defined(my $command = shift @args)) {
+
+    # Number
+    ($results = c($results->[$command])) and next if $command =~ /^\d+$/;
+
+    # Text
+    return _say($results->map('text')->each) if $command eq 'text';
+
+    # All text
+    return _say($results->map('all_text')->each) if $command eq 'all';
+
+    # Attribute
+    return _say($results->map(attr => $args[0] // '')->each) if $command eq 'attr';
+
+    # Unknown
+    die qq{Unknown command "$command".\n};
+  }
+
+  _say($results->each);
+}
+
+1;
+
+=encoding utf8
+
+=head1 NAME
+
+Mojolicious::Command::get - Get command
+
+=head1 SYNOPSIS
+
+  Usage: APPLICATION get [OPTIONS] URL [SELECTOR|JSON-POINTER] [COMMANDS]
```

Comment:
An example with `-c` would be nice here, too:

```
mojo get -H 'Content-Type: application/json' -c '{"hello":"world"}' /
```

It would be worth showing this example here even tho the example doesn't use the JSON string.  Also, does `-c` do anything with a default GET request, or does/can `-c` automatically default to POST?